java-agent/src/main/java/com/signalfx/tracing/examples/javaagent/App.java

```diff
@@ -1,89 +1,120 @@
+        // Redis set action (using the redis-jedis auto-instrumentation).  If you
+        // did not have this root span that is manaully created, you would instead
+        // see two independent traces for the HTTP get and the Redis set.
+        // The Java agent sets the GlobalTracer instance before the application's main method is called.
+        Span span = GlobalTracer.get().buildSpan("fetch-and-set").start();
+        try (Scope sc = GlobalTracer.get().scopeManager().activate(span, false)) {
+            String respBody;
+            try {
+                Response response = httpClient.newCall(request).execute();
+                respBody = response.body().string();
+            } catch (IOException e) {
+                System.out.println("Error: " + e);
+                return;
+            }
 
-    Jedis redisClient = new Jedis("localhost");
+            setInRedisWithThread(url, respBody);
+        } catch (Throwable e) {
+            Tags.ERROR.set(span, true);
+            span.log(Collections.singletonMap(ERROR_OBJECT, e));
+            System.out.println(e);
+        } finally {
+            span.finish();
+        }
 
-    // This span acts as a root span that envelops the sets of spans generated
-    // by both the HTTP call (using the OKHttp auto-instrumentation) and the
-    // Redis set action (using the redis-jedis auto-instrumentation).  If you
-    // did not have this root span that is manaully created, you would instead
-    // see two independent traces for the HTTP get and the Redis set.
-    Span span = tracer.buildSpan("fetch-and-set").start();
-    try (Scope sc = tracer.scopeManager().activate(span, false)) {
-      String respBody;
-      try {
-        Response response = client.newCall(request).execute();
-        respBody = response.body().string();
-      } catch (IOException e) {
-        System.out.println("Error: " + e);
-        return;
-      }
 
-      redisClient.set(url, respBody);
-    } catch (Throwable e) {
-      Tags.ERROR.set(span, true);
-      span.log(Collections.singletonMap(ERROR_OBJECT, e));
-      System.out.println(e);
-    } finally {
-      span.finish();
+        try {
+            executor.shutdown();
+            executor.awaitTermination(3000, TimeUnit.MILLISECONDS);
+            redisClient.close();
+            // Give the tracer time to flush the spans in the sender thread.  In
+            // long-running apps this is generally unnecessary but unfortunately there is no
+            // OpenTracing interface method for closing/stopping a tracer.
+            Thread.sleep(2000);
+        } catch (InterruptedException e) {
+            return;
+        }
     }
 
-    redisClient.close();
+    /*
+     * This annotation will generate a span that wraps the auto-generated Redis SET command span that is
+     * run within a separate thread.  This is a fairly
+     * trivial example that doesn't add much value but at least shows the use of the Trace annotation.
```

Comment:
Might be good to show accessing the current span (that the `@Trace` generates) to add metadata tags to it in the method's body.